lib/index.js

```diff
@@ -211,14 +215,31 @@
             originCallback = options.origin;
           } else if (options.origin) {
             originCallback = function (origin, cb) {
-              cb(null, options.origin);
+              var isAllowed = (!options.origin || options.origin === '*' ||
+                isOriginAllowed(origin, options.origin));
+              // When a user-specified handler returns `false`, we skip applying
+              // the CORS headers and run the request handler (see below). But
+              // the default handler should apply the headers even in case of
+              // a mismatch unless `mismatchContinue === false`.
```

Comment:
I avoid simply returning `cb(null, isAllowed ? options.origin : false)` because [previously](https://github.com/expressjs/cors/pull/110/files#diff-6d186b954a58d5bb740f73d84fe39073L220), a response of `false` would cause `next` to be executed _without_ applying the CORS headers. I'd like the behavior to be unified between this default handler and user-provided handlers but I also don't want to cause breaking changes. I added documentation for this behavior of user-provided handlers [here](https://github.com/expressjs/cors/pull/110/files#diff-04c6e90faac2675aa89e2176d2eec7d8R213).